storage/interface.go

```diff
@@ -92,6 +92,29 @@ type Appender interface {
 }
 
 // SeriesSet contains a set of series.
+// It's caller responsibility to ensure it is always emptied (`Next()` is invoked until false)
+// or error is return from `Err()`. This ensures that all used resources are released.
+//
+// Example usage:
```

Comment:
I hope this was a rule for this interface all the time. Otherwise there is no trival way to implement pure streamed iterators with this that will properly release resources (e.g http body which holds few go routines and TCP conn until you read all and close!). Any thoughts? I feel like this caller responsibility is sane and was that from very beginning. @fabxc ?